common/setups/rasr/onnx_precomputed_hybrid_system.py

```diff
@@ -0,0 +1,167 @@
+    onnx_flow.config[onnx_fwd].session.intra_op_num_threads = cpu
+
+    return onnx_flow
+
+
+class OnnxPrecomputedHybridSystem(HybridSystem):
+    """
+    System class for hybrid systems that train PyTorch models and export them to onnx for recognition. The NN
+    precomputed hybrid feature scorer is used.
+    """
+
+    def nn_recognition(
+        self,
+        name: str,
+        returnn_config: returnn.ReturnnConfig,
+        checkpoints: Dict[int, Union[returnn.Checkpoint, returnn.PtCheckpoint]],
+        train_job: Union[returnn.ReturnnTrainingJob, returnn.ReturnnRasrTrainingJob],
+        prior_scales: List[float],
+        pronunciation_scales: List[float],
+        lm_scales: List[float],
+        optimize_am_lm_scale: bool,
+        recognition_corpus_key: str,
+        feature_flow_key: str,
+        search_parameters: Dict,
+        lattice_to_ctm_kwargs: Dict,
+        parallelize_conversion: bool,
+        rtf: int,
+        mem: int,
+        epochs: Optional[List[int]] = None,
+        quantize_dynamic: bool = False,
+        needs_features_size: bool = True,
+        acoustic_mixture_path: Optional[tk.Path] = None,
+        nn_prior: bool = True,
+        **kwargs,
+    ):
+        """
+        Run recognition with onnx export and precomputed hybrid feature scorer.
+        """
+        with tk.block(f"{name}_recognition"):
+            recog_func = self.recog_and_optimize if optimize_am_lm_scale else self.recog
+
+            feature_flow = self.feature_flows[recognition_corpus_key]
+            if isinstance(feature_flow, Dict):
+                feature_flow = feature_flow[feature_flow_key]
+            assert isinstance(
+                feature_flow, rasr.FlowNetwork
+            ), f"type incorrect: {recognition_corpus_key} {type(feature_flow)}"
+
+            epochs = epochs if epochs is not None else list(checkpoints.keys())
+
+            for pron, lm, prior, epoch in itertools.product(pronunciation_scales, lm_scales, prior_scales, epochs):
+                assert epoch in checkpoints.keys()
+
+                onnx_job = OnnxExportJob(
+                    returnn_config=returnn_config,
+                    checkpoint=checkpoints[epoch],
+                    returnn_root=self.returnn_root,
+                    returnn_python_exe=self.returnn_python_exe,
+                )
+                onnx_job.add_alias(f"export_onnx/{name}/epoch_{epoch}")
+                onnx_model = onnx_job.out_onnx_model
+
+                io_map = {"features": "data", "output": "classes"}
+                if needs_features_size:
+                    io_map["features-size"] = "data_len"
+                onnx_flow = make_precomputed_hybrid_onnx_feature_flow(
+                    onnx_model=onnx_model,
+                    io_map=io_map,
+                    cpu=kwargs.get("cpu", 1),
+                )
+                flow = add_tf_flow_to_base_flow(feature_flow, onnx_flow, tf_fwd_input_name="onnx-fwd-input")
+
+                if nn_prior:
```

Comment:
I think this check rather would construct a prior file which is given to the precomputed scorer. 
So I think we can do smth. like:
```
                    scorer = rasr.PrecomputedHybridFeatureScorer(
                        prior_mixtures=acoustic_mixture_path,
                        priori_scale=prior,
                        prior_file=ComputeNNPriorJob(...).out_prior_file if nn_prior else None
                    )

Reply:
Yes. But I didn't use that so far and you were using a custom `ReturnnForwardComputePriorJob`, so I thought we might first add that or agree on a common way to compute the priors and then we can add it here.